### ISSUE TITLE
Repacking serveral animations into a .pap file?

### ISSUE THREAD

**Issue by Zimmerson** [ilmheg/MultiAssist#2]
It's not exactly an issue per se, (probably more of a feature request) but... I can't create a .pap file that take several of my edited animation.

For example:

I'm trying to reimport the move_a.pap file that has all the different movements.
![image](https://user-images.githubusercontent.com/24922641/165781792-ffde8c1e-9fbb-4a67-b1b1-7c20ae82dc59.png)

I export them out into their own .FBX files. 
![image](https://user-images.githubusercontent.com/24922641/165782383-734d5735-8f1c-4a1a-a208-c3aca673ed1a.png)

I then do the edits and then upload them as .pap files. 
![image](https://user-images.githubusercontent.com/24922641/165782507-3c6af660-e605-4355-82aa-a7799ec138a4.png)

(Ignore the idle and damage idle)

I then import them one by one into the Raw importer, but it just seems to shit itself. Like - I'll upload all of them, the last one being sprint. All of them will be broken (without my animations), whilst the sprint animation is perfectly fine.

If I was to then import another one, like the normal forwards run, the run forward would work, but then the sprint would stop working.

I'm at a bit of a loss right now. Do you know anything?

**Comment by Zimmerson** [comment]
https://github.com/0ceal0t/Dalamud-VFXEditor/wiki/Using-Facial-Expressions 

From the looks of this - the tool by 0ceal0t is able to handle importing several .paps at once. I don't fully know whether it does what it does without checking it out more, but people say that this tool can interract and update individual animations within a .pap file. 

What're your thoughts?

**Comment by Zimmerson** [comment]
Yeah - I've tried using the HEX code to change things - I just don't know enough as to how to fix the problem. I think that's the biggest setback is the several anims in 1 .pap file. 

If you could reverse engineer the extract, I think it'd be doable...

**Comment by ilmheg** [comment]
Just to confirm, are the building upon edited .pap files, or using the original .pap as a base for all animations?

For this, you should be selecting the last exported edited .pap as the selected .pap while importing an animation. If you don't recursively build like this, when importing, TexTools can't tell what's edited and what's not, nor does it have to capability to rebuild .pap files, so you'll be overwriting the entire animation file.

So, using your example, I would import cbnm_01b_lp0 in the original .pap file and export that into, say, `newpap.pap`. Then, when I needed to import cbnm_01f_lp0 into this, I would select newpap.pap as the selected .pap to import into. So, building of the last .pap essentially. When you are done with this process, you will only need to import one .pap file into the game. You can export over newpap.pap each time if you don't mind destructive editing, to save some time.

I can see how this mightn't be ideal however, and I'll consider options to make this a smoother process, like mass importing .fbx and the like, or having a .pap merger.

If this is what you are doing however, I'll review some exports. There shouldn't be any trace of the original animations in exported .pap files.

**Comment by ilmheg** [comment]
I can confirm that recursively repacking edited files functions as intended, and does not retain any information pertaining to the original animations. I'll take measures to make this process clearer

Next release will likely be v0.3.0 with skeleton tools unless something critical pops up, so I'll consider the following for that release:
- Importing multiple .fbx(/.hkx) into one .pap at once.
- .pap merging (Essentially in-app quick animation swapping, albeit with the selection of multiple .paps possible)
- Extract all animations in a .pap at once (Saves time!)

.pap merging will warrant its own tab, but I'll adjust the repacking tab to be able to import multiple edited animations in one repacking action.

**Comment by Zimmerson** [comment]
Apologies if I'm making this/things difficult for you 🙏 

I use the original .pap as a base, so for example, I take all my edited animations and import them one by one to the move_a.pap file.

I'll give it a go where I import cbnm_01b_lp0.fbx into the move_a.pap, then export that as move_a_1.pap, then take  cbnm_01f_lp0.fbx and import that into move_a_1.pap and call it move_a_2.pap, so on and so forth, right?

If so - I never thought about that. I was just doing the straight forward thing. I didn't realise that TexTools couldn't handle partial animation imports that would recognise which bits being replaced, so I'll try your method. 

The ideas for v0.3.0 has got me excited thogh, I'm looking forward to what you'll do, as it's been impressive to see where this tool has gone, and it'll be exciting to see where you'll take it 👍 

I'll report back with my results.

**Comment by ilmheg** [comment]
I hope it works out! And no need to apologize for anything, seeing how people use software is essential to creating better software! It's super easy to have blind spots in my perception of how people will use it, so getting feedback is always a great thing.

And yep, moving from .pap to .pap is the right idea, you should also be able to just keep using the same .pap too, but creating a new .pap will naturally be safer.

**Comment by Zimmerson** [comment]
Can confirm - it does work!

I totally didn't know you could do this. I was utterly stumped last night, feeling as if I couldn't do a multiple .pap animation. Now that I know of this - ohoho boy I can do what I want now!

Thanks for notifying me of that, my move_a.pap animations work now :D
![image](https://user-images.githubusercontent.com/24922641/165916317-af40806b-efdd-4020-a3d3-7ac19ee7853a.png)
Cheeky Screenshot of my Tail when jumping! IT ISNT STIFF!

All i've got left is to work on the individual poses and it's done! (Although that's like... 130 animations 😂)
